app/module/configurator.go

```diff
@@ -91,3 +125,35 @@ func (c configurator) runModuleMigrations(ctx sdk.Context, moduleName string, fr
 
 	return nil
 }
+
+// the server wrapper wraps the pbgrpc.Server for registering a service but
+// includes logic to extract all the sdk.Msg types that the service declares
+// in its methods and fires a callback to add them to the configurator.
+// This allows us to create a map of which messages are accepted across which
+// versions
+type serverWrapper struct {
+	addMessages func(msgs []string)
+	msgServer   pbgrpc.Server
+}
+
+func (s *serverWrapper) RegisterService(sd *grpc.ServiceDesc, v interface{}) {
+	msgs := make([]string, len(sd.Methods))
+	for idx, method := range sd.Methods {
```

Comment:
are messages always 1:1 with methods?

Reply:
Yup, each method only ever has one message which is the request argument